packages/functions/src/index.ts

```diff
@@ -0,0 +1,98 @@
+import * as crypto from 'crypto';
+
+import * as admin from 'firebase-admin';
+import * as functions from 'firebase-functions';
+import * as nodemailer from 'nodemailer';
+
+admin.initializeApp();
+
+const SENDER_EMAIL = process.env.GMAIL_SENDER;
+const GMAIL_PASSWORD = process.env.GMAIL_PASSWORD;
+
+if (!SENDER_EMAIL || !GMAIL_PASSWORD) {
+  throw new Error('Sender email and password must be set as environment variables.');
+}
+
+const transporter = nodemailer.createTransport({
+  host:   'smtp.gmail.com',
+  port:   465,
+  secure: true,
+  auth:   {
+    user: SENDER_EMAIL,
+    pass: GMAIL_PASSWORD,
+  },
+});
+
+export const sendOTP = functions.https.onCall(
+  async (data: { email: string }, _context) => {
+    const { email } = data;
+
+    const otp = crypto.randomInt(100000, 999999).toString();
+
+    const otpDoc = admin.firestore().collection('otps').doc(email);
+    await otpDoc.set({
+      otp,
+      createdAt: admin.firestore.FieldValue.serverTimestamp(),
+      expiresAt: admin.firestore.Timestamp.fromDate(
+        new Date(Date.now() + 10 * 60 * 1000)
+      ), // 10 minutes expiration
+    });
+
+    const now = new Date();
+    const formattedDate = now.toISOString().replace(/T/, ' ').replace(/\..+/, ' Z');
+
+    const emailContent = `Hello,<br><br>We received a request to sign in to NEAR Onboarding using this email address, at ${formattedDate}. If you want to sign in with your ${email} account, use this code:<br><br>${otp}<br><br>If you did not request this link, you can safely ignore this email.<br><br>Thanks,<br><br>Your NEAR Onboarding team`;
+
+    await transporter.sendMail({
```

Comment:
Hey @esaminu, what happens to the  otp `doc` if email sending fails? Or can we store it only when email sending succeeds?